### PR TITLE
Fixes #35432 - Use Rails 6.1 defaults

### PR DESCRIPTION
**Pull request by ekohl**
Rails has introduced [config.load_defaults](https://guides.rubyonrails.org/configuring.html#versioned-default-values) to load the defaults of a certain version. Currently this is not called at all, which means using Rails < 5.0 defaults. Changing this brings Foreman more in line with the Rails recommended defaults.

Right now this is a draft since I'm a bit uncertain of all the implications.

### CODE REVIEW

**Comment by ekohl** [comment]
Quite certain test failures are related. Will need to dive into them.

**Comment by ekohl** [comment]
I've split it into multiple commits so I can dive into it and see where things break. Currently running tests on these.

File: app/models/host/base.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ class Base < ApplicationRecord
     friendly_id :name
 
     validates_lengths_from_database
-    belongs_to :model, :name_accessor => 'hardware_model_name'
+    belongs_to :hardware_model, class_name: 'Model', foreign_key: 'model_id'
```
Comment (ekohl) [inline app/models/host/base.rb]: This is not the same - the `model` and `model=` methods are still used. It's also part of the REST API.

File: app/models/concerns/has_many_common.rb [hunk 2]
```diff
@@ -73,6 +73,9 @@ def has_many_names_for(association, options)
     #### belongs_to ####
     def belongs_to(*args)
       options = args.last.is_a?(Hash) ? args.last : {}
+      if options.key?(:name_accessor)
+        Foreman::Deprecation.deprecation_warning('3.6', 'Passing name_accessor to belongs_to is deprecated, use native options instead')
```
Comment (ezr-ondrej) [inline app/models/concerns/has_many_common.rb]: can we define the methods and throw the deprecation warning there - in the usage of the methods?

Comment (ekohl) [reply app/models/concerns/has_many_common.rb]: My reasoning is that I want to get rid of this `belongs_to` definition altogether. Ideally only Rails defines it. Though after https://github.com/theforeman/foreman/pull/9332 it needs a rebase anyway.

**Comment by ekohl** [comment]
Rebased on #9332 now. Still need to dive into the failures.

**Comment by ekohl** [comment]
For encryption I think it the requirements on the key may have been tightened. The key looks to be md5 (they are always 32 characters):
https://github.com/theforeman/foreman/blob/06600bcca9025390a004413700deda4fa601837e/test/unit/encryptable_test.rb#L17
Perhaps updating this to a stronger key is sufficient.

As for the interface validation errors, I don't know. I wish the errors were less vague.

**Pull request closed without merging**